diff -urNp mc-4.8.20-orig/src/usermenu.c mc-4.8.20/src/usermenu.c
--- mc-4.8.20-orig/src/usermenu.c	2017-11-18 19:45:59.011683696 +0200
+++ mc-4.8.20/src/usermenu.c	2017-11-27 12:29:21.593998903 +0200
@@ -35,6 +35,7 @@
 #include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
+#include <limits.h>
 
 #include "lib/global.h"
 #include "lib/fileloc.h"
@@ -1119,9 +1120,13 @@ user_menu_cmd (const WEdit * edit_widget
             /* insert all the items found */
             for (i = 0; i < menu_lines; i++)
             {
-                p = entries[i];
-                LISTBOX_APPEND_TEXT (listbox, (unsigned char) p[0],
-                                     extract_line (p, p + MAX_ENTRY_LEN), p, FALSE);
+                p = g_locale_from_utf8 (entries[i], -1, NULL, NULL, NULL);
+                if (p)
+                {
+                    LISTBOX_APPEND_TEXT (listbox, (unsigned char) p[0],
+                                         extract_line (p, p + MAX_ENTRY_LEN * MB_LEN_MAX), p, FALSE);
+                    g_free (p);
+                }
             }
             /* Select the default entry */
             listbox_select_entry (listbox->list, selected);
