--- gtk3-3.24.5/gtk/ui/gtkfilechooserbutton.ui.orig	2019-02-19 12:00:49.629990042 +0200
+++ gtk3-3.24.5/gtk/ui/gtkfilechooserbutton.ui	2019-02-19 12:12:38.609987008 +0200
@@ -39,16 +39,6 @@
               </packing>
             </child>
             <child>
-              <object class="GtkSeparator" id="separator1">
-                <property name="visible">1</property>
-                <property name="orientation">vertical</property>
-              </object>
-              <packing>
-                <property name="fill">0</property>
-                <property name="position">2</property>
-              </packing>
-            </child>
-            <child>
               <object class="GtkImage" id="open_file_icon">
                 <property name="visible">1</property>
                 <property name="icon-name">document-open-symbolic</property>
--- gtk3-3.24.5/gtk/ui/gtkfontbutton.ui.orig	2019-02-19 12:00:49.629990042 +0200
+++ gtk3-3.24.5/gtk/ui/gtkfontbutton.ui	2019-02-19 12:11:36.321987275 +0200
@@ -22,15 +22,6 @@
             <property name="visible">1</property>
             <property name="no-show-all">1</property>
             <child>
-              <object class="GtkSeparator" id="separator">
-                <property name="visible">1</property>
-                <property name="orientation">vertical</property>
-              </object>
-              <packing>
-                <property name="fill">0</property>
-              </packing>
-            </child>
-            <child>
               <object class="GtkLabel" id="size_label">
                 <property name="visible">1</property>
                 <property name="label">14</property>
